### PR TITLE
add spread operator support to JavaScript AST

### PR DESCRIPTION
**Pull request by TyOverby**
This PR adds support for the Javascript spread operator in function call position.

This is necessary for linking with any Javascript that uses this operator.

### CODE REVIEW

File: compiler/lib/generate.ml [hunk 1]
```diff
@@ -309,7 +309,7 @@ let rec constant_rec ~ctx x level instrs =
   | String s ->
       let e = Share.get_string str_js s ctx.Ctx.share in
       let p = Share.get_prim (runtime_fun ctx) "caml_new_string" ctx.Ctx.share in
-      J.ECall (p, [ e ], J.N), instrs
+      J.ECall (p, [ e, `Not_spread ], J.N), instrs
```
Comment (hhugo) [inline compiler/lib/generate.ml]: Could we maybe introduce a helper function
```ocaml
let ecall f args loc = J.ECall (f, List.map args ~f:(fun x -> x, `Not_spread), loc)
```

**Comment by TyOverby** [comment]
I ran `make fmt` which was apparently a mistake.  I'm using the same version of ocamlformat (`opam install -y ocamlformat.0.12`) but it clearly produced different output. 

Any idea what's going on here?

**Comment by hhugo** [comment]
The CI is green on your last commit, did you figure out what was wrong ?

**Comment by TyOverby** [comment]
@hhugo I have no idea what I did.  But did you look at the diff?  There's a bunch of whitespace changes.

**Comment by hhugo** [comment]
What whitespace diff are you talking about ?

**Comment by TyOverby** [comment]
Wait, I'm looking at the wrong diff.  Sorry; all is well!

**Pull request merged**